src/Symfony/Component/Notifier/Bridge/Smsc/SmscTransport.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Smsc;
+
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class SmscTransport extends AbstractTransport
+{
+    protected const HOST = 'smsc.ru';
+
+    private $login;
+    private $password;
+    private $from;
+
+    public function __construct($username, $password, $from, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->login = $username;
+        $this->password = $password;
+        $this->from = $from;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('smsc://%s?from=%s', $this->getEndpoint(), (string) $this->from);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $query = [
+            'login' => $this->login,
+            'psw' => $this->password,
+            'sender' => (string) $this->from,
+            'phones' => $message->getPhone(),
+            'mes' => $message->getSubject(),
+            'fmt' => 3, // response as JSON
+            'charset' => 'utf-8',
+            'time' => '0-24',
+        ];
+
+        $endpoint = sprintf('https://%s/sys/send.php', $this->getEndpoint());
+        $response = $this->client->request('GET', $endpoint, ['query' => $query]);
+
+        $result = $response->toArray(false);
+        if (\array_key_exists('error', $result)) {
+            throw new TransportException(sprintf('Unable to send the SMS: code = %d, message = "%s".', $result['error_code'], $result['error']), $response);
+        }
+
+        $sentMessage = new SentMessage($message, (string) $this);
+        $sentMessage->setMessageId((string) $result['id']);
```

Comment:
Maybe be extra safe by: 

```suggestion
        $sentMessage->setMessageId((string) $result['id'] ?? '');
```

Reply:
@Nyholm The cast operator has a higher precedence than `??` unfortunately.

```suggestion
        $sentMessage->setMessageId((string) ($result['id'] ?? ''));
```

Reply:
Oh. The 1:30 AM @derrabus is way more observant to details than I am. 

=)

Well done.

Reply:
Added this failsafe